crates/risc0/src/host.rs

```diff
@@ -44,7 +43,7 @@ pub struct Risc0BonsaiHost {
 
 impl Risc0BonsaiHost {
     /// Create a new Risc0Host to prove the given binary.
-    pub fn new(image_id: Digest, elf: Vec<u8>, ledger_db: LedgerDB) -> Self {
+    pub fn new(image_id: Digest, ledger_db: LedgerDB) -> Self {
```

Comment:
image_id is irrelevant at this point if we are passing elfs on `run`

Reply:
```
        receipt.verify(self.image_id)?;
```
In host, we use the image ID to verify receipt apparently.

Reply:
yes but image id can be calculated from the elf passed wherever needed.